keyboards/opendeck/32/rev1/rev1.c

```diff
@@ -63,27 +63,28 @@ const is31_led PROGMEM g_is31_leds[RGB_MATRIX_LED_COUNT] = {
     // clang-format on
 };
 
-#define P(x,y) {(x*224 / (MATRIX_COLS - 1)), (y*64 / (MATRIX_ROWS - 1))}
```

Comment:
Please don't do this. It breaks configurability by hard coding in the matrix size, meaning that porting to different sizes of the grid becomes annoying.

Reply:
The problem here is that the CLI can't understand the values in g_led_config when they use macros, and changing the CLI to evaluate them would introduce far too much complexity, and breaks the two-way data-driven mapping.

Reply:
Can't the CLI just run CPP (the C Pre Processor) first and then run on the output of that?

Reply:
That was attempted in the work that added `info.json` support, but it was non trivial due to the variables introduced in the build process. Support for parsing `.c` files is not intended long term, and was introduced as mostly a "best effort" to aid the migration process. Instead **all** config will eventually be the `info.json` format, which right now is still intended to be just numeric values. At a point in the future, this will become mandatory for all keyboards in the repo, and those added in PRs.

Reply:
I'm not familiar with the format of the `info.json` file, but I imagine that adding a optional rational x and y scale factors to the `info.json` and multiplying that out in the parsing should be doable. That way the macro here could be removed, leaving the indexes as is and adding the scale factor in the `info.json`.

I don't know fully what the rollout plan of this is, nor do I have the time to get into it. But my spidey sense is tingling here and I get the feeling that this PR is premature. Can the `info.json` format be completed first (hopefully with the above feature) and then the boards migrated directly to that format without this intermediate step that breaks configurability and adds maintenance pain?

Reply:
To us, keeping this code in as is, is maintenance pain. Its causing issue in the short term. Extensions to the `info.json` format can happen if someone puts the work in, but that unlikely to be a blocker on this PR getting merged "as is".

Reply:
Data-driven is the way to go at this point -- as @zvecr said we'll welcome future PRs to simplify the definitions, but for now the intended direction is to move away from specifying them in code.

If we look at all the places that may be considered "premature" then we'd have no progress until we "big bang" everything at once, generating even more friction. We've chosen to chip away and iterate over time as a long-term mitigation strategy instead.

Reply:
I realize running a large projects with many contributors is hard. But I disagree with how heavyhandedly this was handled so no need to bother CCing me on future PRs, if you're just going to ignore the comments and go ahead with your change anyway, I'd rather not waste my time.

I also do not accept that features of my project are broken and maintenance burden is applied unilaterally. The code is now yours to maintain.

Reply:
I have to say I don't really understand the benefit of using a macro here. If you want to reuse it for multiple boards with varying layouts (say, an 8x8 version)... you still have to edit the `g_led_config` to add the extra LED indexes/positions/flags anyway.

Additionally, and perhaps more importantly, this specific section of `g_led_config` is *not supposed to be tied to the electrical matrix dimensions*. It's purely for assigning physical X/Y positions to each LED (which may or may not even be associated with a key, as you can flag LEDs as underglow only or even indicator only).

So while this solution works for your particular case, a simple ortholinear layout with a matrix that matches it, and only per-key LEDs, it doesn't necessarily work for every other board.